### ISSUE TITLE
Augment benchmarks/vectorization_strategy_benchmark.py to include XLA compilation

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#165]

**Comment by bhack** [comment]
Check:
https://github.com/keras-team/keras-cv/pull/161/files#r820872659

**Comment by bhack** [comment]
It seems that in faster `jit_compiled` version all the `tf.cast` we have are still quite relevant

![immagine](https://user-images.githubusercontent.com/1710528/157131716-9bc840cc-5ff6-4ba5-a90e-6e2d6a7f5988.png)


![immagine](https://user-images.githubusercontent.com/1710528/157132032-467636b7-6fd6-47f2-9834-e8bc387d8102.png)

**Comment by bhack** [comment]
@qlzh727 What do you think about adding an extra parameter to the base class for `jit_compile`?
https://github.com/keras-team/keras/blob/master/keras/layers/preprocessing/image_preprocessing.py#L413-L414

So that we could optionally use
```python 
f = def_function.function(self._augment, jit_compile=True)
self._map_fn(f, inputs)
```

**Comment by LukeWood** [comment]
> @qlzh727 What do you think about adding an extra parameter to the base class for `jit_compile`? https://github.com/keras-team/keras/blob/master/keras/layers/preprocessing/image_preprocessing.py#L413-L414
> 
> So that we could optionally use
> 
> ```python
> f = def_function.function(self._augment, jit_compile=True)
> self._map_fn(f, inputs)
> ```

I personally would support a jit_compile attribute on the BaseImageAugmentationLayer.

so in a constructor you'd do:

jit_compile = True 

and it would work pass it to the tf.function wrapping `call`.

**Comment by bhack** [comment]
We need to care about massively closing bug a we lost interesting threads. This is now at https://github.com/keras-team/keras/issues/16403

**Comment by LukeWood** [comment]
> We need to care about massively closing bugs a we lose interesting threads. This is now at https://github.com/keras-team/keras/issues/16403

This is a distinct issue.  I do not think benchmarks do not need this as the benchmarks existed to guide our direction. 

you are right, BaseImageAugmentationLayer should support jit_compile, but the benchmarks do not need this so I closed this issue.

Closing bug is part of maintaining a clean project state.

**Comment by bhack** [comment]
> Closing bug is part of maintaining a clean project state.

Yes but if there are still interesting parts emerged in the thread you need to spinoff that part on a new ticket or it will be lost forever.
